### PR TITLE
S16

### PR DESCRIPTION
**Pull request by devwistkey**
@tralkan

### CODE REVIEW

File: components/SearchBar/index.tsx [hunk 1]
```diff
@@ -20,11 +20,14 @@ import { updateResolvedInfo } from '../../src/store/resolvedInfo'
 import { updatePrev } from '../../src/store/router'
 import smartContract from '../../src/utils/smartContract'
 import toastTheme from '../../src/hooks/toastTheme'
+import useArConnect from '../../src/hooks/useArConnect'
 
 function Component() {
     const zcrypto = tyron.Util.default.Zcrypto()
     const Router = useRouter()
+    const { connect } = useArConnect()
```
Comment (tralkan) [inline components/SearchBar/index.tsx:28]: is this regarding https://www.notion.so/ssiprotocol/Feedback-4-1846f6d4397848c189828e6d0e7f3d2f#95aaa73a396c41e0b94828879def9094 @devwistkey ?

File: components/SSI/AddFunds/index.tsx [hunk 2]
```diff
@@ -242,7 +243,83 @@ function Component(props: InputType) {
...
+                )
+                const get_services = await getSmartContract(
+                    init_addr,
+                    'services'
+                )
+                const services = await tyron.SmartUtil.default.intoMap(
+                    get_services.result.services
+                )
+                token_addr = services.get(id)
+                const balances = await getSmartContract(token_addr, 'balances')
```
Comment (tralkan) [inline components/SSI/AddFunds/index.tsx:269]: this code gets run twice, also in wallet info? @devwistkey

**Review by tralkan — CHANGES_REQUESTED** [review]

File: components/SearchBar/index.tsx (hunk 1 above)
Comment (devwistkey) [reply components/SearchBar/index.tsx:28]: Yes, you're right @tralkan

File: components/SSI/AddFunds/index.tsx (hunk 2 above)
Comment (devwistkey) [reply components/SSI/AddFunds/index.tsx:269]: Yes, on wallet info we need it to show balance. On this file we need it to check the balance compare to the input amount @tralkan

**Review by tralkan — APPROVED** [review]

**Pull request merged**